client/components/place/place-list/place-list.component.js

```diff
@@ -79,6 +80,222 @@ angular.module('placeList', ['filterMapType', 'popularTracks', 'ngAnimate'])
+          newPointsForTrack.splice(pointIndexInArray, ctrl.newTrackPoints.length - pointIndexInArray);
+          placesForTrack.splice(pointIndexInArray, ctrl.newTrackPoints.length - pointIndexInArray);
+          ctrl.newTrackPoints.splice(pointIndexInArray, ctrl.newTrackPoints.length - pointIndexInArray);
+          removedPoints.forEach(function(point) {
+            if (point) {
+              map.removeLayer(point);
+            }
+          });
+          if (newPointForTrack) {
+            map.removeLayer(newPointForTrack);
+          }
+          map.removeLayer(newTrack);
+          addNewTrackOnMap(placesForTrack);
+        };
+
+        ctrl.createNewTrack = function(form) {
+          var checkActiveType;
+          var newPointsCounter = 0;
+          var counterByNewPoints = 0;
+          if (form.$valid) {
+            ctrl.newTrackObject.owner = ctrl.user._id;
+            ctrl.newTrackPoints.forEach(function(point) {
+              if (!point[0]._id) {
+                newPointsCounter++;
+              }
+            });
+            if (newPointsCounter == 0) {
+              addNewTrackIntoDB(ctrl.newTrackPoints, form);
+            } else {
+              ctrl.newTrackPoints.forEach(function(point, index) {
+                var newPoints = [];
+                if (!point[0]._id)  {
+                  newPoints.push(point[0]);
+                  Restangular.oneUrl('location', 'https://nominatim.openstreetmap.org/reverse?format=json&lat=' + point[0].location.coordinates[1] +
+                  '&lon=' + point[0].location.coordinates[0] + '&addressdetails=0&zoom=10').get().then(function(result) {
+                    point[0].address = result.display_name;
+                    Place.post(point[0]).then(function(response) {
+                      counterByNewPoints++;
+                      point[0]._id = response.record._id;
+                      checkActiveType = angular.element('.' + point[0].type + ' span');
+                      if (checkActiveType.hasClass(constants.checkedSpanClass)) {
+                        placesOnMap.showPlaces(newPoints);
+                      } else {
+                        ctrl.checkType(point[0].type);
+                      }
+                      if (counterByNewPoints == newPointsCounter) {
+                        addNewTrackIntoDB(ctrl.newTrackPoints, form);
+                      }
+                    });
+                  });
+                }
+              });
+            }
+          }
+        };
+
+        function addNewTrackIntoDB(array, form) {
+          array.forEach(function(item) {
+            ctrl.newTrackObject.places.push(item[0]._id);
+          });
+          Track.post(ctrl.newTrackObject).then(function(response) {
+            var checkActiveType = angular.element('.' + ctrl.newTrackObject.type + ' span:last-child');
+            if (checkActiveType.hasClass(constants.checkedSpanClass)) {
```

Comment:
I duplicates this function because when user creates new track with type wich is already chacked, I must only show this track on map without getting all tracks from db. For this reason I can use `function placesOnMap.showTracks`, but this function expects to recieve track object with location field (which populates on server side), so now I can't use this function, but for showing new track on map I just calls function first time for removing all track, and calls it second time for getting all tracks (including new track) from db.
I will be able to use `placesOnMap.showTracks` after adding `location` field in `track `model.